rust_src/src/numbers.rs

```diff
@@ -294,10 +295,13 @@ pub fn number_or_marker_p(object: LispObject) -> bool {
 pub fn random(limit: LispObject) -> LispObject {
     let mut rng = RNG.lock().unwrap();
     if limit.is_t() {
-        *rng = StdRng::new().unwrap();
+        *rng = StdRng::from_entropy();
     } else if let Some(s) = limit.as_string() {
-        let values: Vec<usize> = s.as_slice().iter().map(|&x| x as usize).collect();
-        rng.reseed(&values);
```

Comment:
Do you need the `seed` array here? Can you not pass in a 32 element slice from `values` like `&values[0..32]`?

Reply:
As I recall, compiler complained that `from_seed()` had expected `[u8; 32]` but found `&[u8]`. I couldn't get better way to cast these types...

Reply:
I see what you mean. Oh well.